From 6a312085a713a2a048e5f40b3e500024f88c9b40 Mon Sep 17 00:00:00 2001
From: Konrad Materka <materka@gmail.com>
Date: Mon, 22 Jun 2020 23:59:00 +0200
Subject: [PATCH] [libdbusmenuqt] Fix memory leak

QAction is not a widged, so it is not possible to set it as a parent of QMenu.
As a result, QMenu assigned to QAction is not automatically deleted when action is removed from menu and/or deleted.
---
 libdbusmenuqt/dbusmenuimporter.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libdbusmenuqt/dbusmenuimporter.cpp b/libdbusmenuqt/dbusmenuimporter.cpp
index 50a8f4d83..cab0a26ed 100644
--- a/libdbusmenuqt/dbusmenuimporter.cpp
+++ b/libdbusmenuqt/dbusmenuimporter.cpp
@@ -414,6 +414,9 @@ void DBusMenuImporter::slotGetLayoutFinished(QDBusPendingCallWatcher *watcher)
             // which can happen when an application completely reloads this menu.
             // When the action is deleted deferred, it is removed from the menu.
             action->deleteLater();
+            if (action->menu()) {
+                action->menu()->deleteLater();
+            }
             d->m_actionForId.remove(id);
         }
     }
